frontend/src/pages/projects/notebook/useRouteForNotebook.ts

```diff
@@ -11,7 +12,11 @@ const useRouteForNotebook = (
 
   React.useEffect(() => {
     let watchHandle;
+    let cancelled = false;
     const watchRoute = () => {
+      if (cancelled) {
+        return;
+      }
       if (notebookName && projectName) {
         getRoute(notebookName, projectName)
           .then((route) => {
```

Comment:
We need cancelling for both `then` and `catch` too -- otherwise there will be react errors.